### ISSUE TITLE
[a11y] Add expanded state in Semantics widget on Android

### ISSUE THREAD

**Issue by GabrieleSuzart** [flutter/flutter#92040]
## Use case

I cannot set expanded or collapsed states for screen readers. This accessibility issue is WCAG's conformance level A, Criterion 4.1.2: Name, Role, Value
WCAG: https://www.w3.org/WAI/WCAG21/Understanding/name-role-value

I would like the `Semantics` widget to have an `expanded` parameter so that it announces the state for VoiceOver and TalkBack.

After creating this parameter in `Semantics`, I recommend putting it in the `Material's Accordion` and similar components.

## Proposal

Analyze the behavior of the `accessibilityState={{expanded: true}}` parameter of React Native and do the same.

https://github.com/facebook/react-native/search?q=expanded

## Extra

W3C Documentation about HTML's aria-expanded: https://www.w3.org/WAI/GL/wiki/Using_aria-expanded_to_indicate_the_state_of_a_collapsible_element

React Native documentation about accessibilityState: https://reactnative.dev/docs/accessibility#accessibilitystate

Example of accordion accessible in react material design: https://mui.com/pt/components/accordion/

**Comment by chunhtai** [comment]
The new Android version 36 added a new getExpandedState getter

**Comment by chunhtai** [comment]
Also the expanded state is already added to the semantics widget https://api.flutter.dev/flutter/semantics/SemanticsProperties/expanded.html

**Comment by kevmoo** [comment]
Waiting on new Android release to address this.

Track https://github.com/flutter/flutter/issues/162141 for web implementation

**Comment by jwlilly** [comment]
@chunhtai Android 16 is out now. I can take this one

**Comment by ksokolovskyi** [comment]
Hi @jwlilly, thanks a lot for your interest, but I am already working on this issue and will soon file a PR.

**Comment by rekire** [comment]
I'm not sure if I run into a merge conflict or something similar, but in my current dev branch has `semantics.dart` an unresolved reference to `SemanticsAction.collapse` and `SemanticsAction.expand`. Is this a known issue? Git blame references this ticket here, so I'm a bit confused.

I just notices that the file is in `bin/cache/pkg/sky_engine/lib/ui/semantics.dart` does that mean that I have a incompatible sky_engine version? How can I fix it?

**Comment by ksokolovskyi** [comment]
> I'm not sure if I run into a merge conflict or something similar, but in my current dev branch has `semantics.dart` an unresolved reference to `SemanticsAction.collapse` and `SemanticsAction.expand`. Is this a known issue? Git blame references this ticket here, so I'm a bit confused.
> 
> I just notices that the file is in `bin/cache/pkg/sky_engine/lib/ui/semantics.dart` does that mean that I have a incompatible sky_engine version? How can I fix it?

Hi @rekire, please make sure that you have merged the latest master into your dev branch. If yes, then try to remove `bin/cache` folder and run `flutter doctor` to download the fresh `sky_engine`. I hope this helps.
